recipes-sourcery/glibc-sourcery/glibc-sourcery.bb

```diff
@@ -149,8 +144,16 @@ bberror_task-install () {
     :
 }
 
-require recipes-external/glibc/glibc-sysroot-setup.inc
-require recipes-external/glibc/glibc-package-adjusted.inc
+linux_include_subdirs = "asm asm-generic bits drm linux mtd rdma sound sys video"
+
+do_install_append () {
+    for dir in ${linux_include_subdirs}; do
+        rm -rf "${D}${includedir}/$dir"
+    done
+    if [ "${baselib}" != "lib" ]; then
+        rmdir --ignore-fail-on-non-empty "${D}${prefix}/lib/locale" "${D}${prefix}/lib"
```

Comment:
@kergoth, I understand that this works as expected. Wouldn't it make more sense if we do this using
```
do_poststash_install_cleanup_append () {
    if [ "${baselib}" != "lib" ]; then
        rmdir --ignore-fail-on-non-empty "${D}${prefix}/lib"
    fi
}
```

As all the cleanups are happening there I believe that would be the right thing to do. I've submitted https://patchwork.openembedded.org/patch/154968/ upstream as well.

Reply:
Seems reasonable to me, I didn't dig deeply into how the locale stash cleanup operates. By all means submit another pull req if that approach seems more reasonable. I'm on vacation at the moment, so haven't been able to connect very often.